File: Programação Orientada a Objetos/Trabalho Prático 2/mavenproject1/src/main/java/Casa.java
public class Casa extends Imovel {
    //Características específicas da Casa
    private int andares;
    private boolean sala_jantar;

    //Construtor
    public Casa (int i, float v, String p, String r, String b, String c, int n, int q, int ba, int a, boolean sj) {
        super(i, v, p, r, b, c, n, q, ba);
        andares = a;
        sala_jantar = sj;
    }

    //Método do printar
    @Override
    public void print() {
        super.print();
        System.out.printf("Andares: %d\n", this.andares);

    }

    //Getters e Setters
    public int getAndares() {
        return andares;
    }
    
    public void setAndares(int a){
        andares = a;
    }
    
    public boolean getSalaJantar() {
        return sala_jantar;
    }
    
    public void setSalaJantar(boolean sj){
        sala_jantar = sj;
    }

}

File: Programação Orientada a Objetos/Trabalho Prático 1/casa.cpp
#include "casa.h"
#include "imovel.h"
 
//Construtor
Casa::Casa(int i, float v, std::string p, std::string r, std::string b, std::string c, int n, int q, int ba, int a, bool sj) : Imovel(i, v, p, r, b, c, n, q, ba){
    andares = a;
    sala_jantar = sj;
}

//Overload do método print
std::ostream& Casa::print(std::ostream& os) const{
    os  << "----\n"
        << "\t" << this->getProprietario() << "\n"
        << "\t" << this->getValor() << "\n"
        << "\t" << this->getQuartos() << "\n"
        << "\t" << this->getRua() << "\n"
        << "\t" << this->getBairro() << "\n"
        << "\t" << this->getCidade() << "\n"
        << "\t" << this->getAndares() << "\n"
        << "----\n";
    return os;
}

//Getters e Setters
int Casa::getAndares() const{
    return andares;
}

void Casa::setAndares(int a){
    andares = a;
}

bool Casa::getSalaJantar() const{
    return sala_jantar;
}

void Casa::setSalaJantar(float sj){
    sala_jantar = sj;
}